packages/contracts/test/invariant/diamond/facets/PoolFacetHandler.sol

```diff
@@ -0,0 +1,266 @@
+     * @param _ethUsdPriceFeed The mock price feed for ETH/USD.
+     * @param _ubiquityPoolFacet The pool facet contract responsible for managing Ubiquity Dollars.
+     * @param _admin The address with admin privileges, used for administrative actions in the pool.
+     * @param _user The user address that interacts with the pool for testing purposes.
+     * @param _curveDollarPlainPool The mocked Curve pool contract used for dollar-based operations.
+     */
+    constructor(
+        MockChainLinkFeed _collateralTokenPriceFeed,
+        MockChainLinkFeed _stableUsdPriceFeed,
+        MockChainLinkFeed _ethUsdPriceFeed,
+        UbiquityPoolFacet _ubiquityPoolFacet,
+        address _admin,
+        address _user,
+        MockCurveStableSwapNG _curveDollarPlainPool
+    ) {
+        collateralTokenPriceFeed = _collateralTokenPriceFeed;
+        stableUsdPriceFeed = _stableUsdPriceFeed;
+        ethUsdPriceFeed = _ethUsdPriceFeed;
+        ubiquityPoolFacet = _ubiquityPoolFacet;
+        admin = _admin;
+        user = _user;
+        curveDollarPlainPool = _curveDollarPlainPool;
+    }
+
+    /**
+     * @notice Manipulates the Ubiquity Dollar price to a value above a set threshold.
+     * @dev This function assumes the new dollar price is within the specified range (greater than 1e18 and less than 2e18).
+     * It then updates the mocked Curve pool parameters and adjusts the collateral ratio in the UbiquityPoolFacet.
+     * @param newDollarPrice The new price for Ubiquity Dollar, expected to be within the range of 1e18 to 2e18.
+     */
+    function setDollarPriceAboveThreshold(uint256 newDollarPrice) public {
+        vm.assume(newDollarPrice > 1e18 && newDollarPrice < 2e18);
+
+        vm.prank(admin);
+        curveDollarPlainPool.updateMockParams(newDollarPrice);
+
+        uint256 reductionFactor = newDollarPrice.sub(1e18).div(1e16);
+        uint256 newCollateralRatio = uint256(1e6).sub(reductionFactor);
+
+        ubiquityPoolFacet.setCollateralRatio(newCollateralRatio);
+    }
+
+    /**
+     * @notice Manipulates the Ubiquity Dollar price to a value below a set threshold.
+     * @dev This function assumes the new dollar price is within the specified range (greater than or equal to 0.5e18 and less than 1e18).
+     * It then updates the mocked Curve pool parameters and adjusts the collateral ratio in the UbiquityPoolFacet accordingly.
+     * @param newDollarPrice The new price for Ubiquity Dollar, expected to be within the range of 0.5e18 to 1e18.
+     */
+    function setDollarPriceBelowThreshold(uint256 newDollarPrice) public {
+        vm.assume(newDollarPrice >= 0.5e18 && newDollarPrice < 1e18);
+
+        vm.prank(admin);
+        curveDollarPlainPool.updateMockParams(newDollarPrice);
+
+        uint256 increaseFactor = uint256(1e18).sub(newDollarPrice).div(1e16);
+        uint256 newCollateralRatio = uint256(1e6).add(increaseFactor);
+
+        ubiquityPoolFacet.setCollateralRatio(newCollateralRatio);
```

Comment:
This line always reverts because `collateralRatio` [can't be greater](https://github.com/ubiquity/ubiquity-dollar/blob/0a230f977ad91bbfb01dd28bc1d2ea77faed5237/packages/contracts/src/dollar/libraries/LibUbiquityPool.sol#L1068) than `1_000_000`. Pls refactor accordingly.

Reply:
The invocation of setCollateralRatio has been completely removed from the Handler